### PR TITLE
[DON'T MERGE] Surface tracker fix background processing

### PR DESCRIPTION
**Pull request by romanroibu**
Addresses #1588

### CODE REVIEW

File: pupil_src/shared_modules/background_helper/base.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Copyright (C) 2012-2019 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import abc
+import typing
```
Comment (pfaion) [inline pupil_src/shared_modules/background_helper/base.py]: Let's use `import typing as T` for consistency.
Also in the other files.

File: pupil_src/shared_modules/background_helper/base.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class EarlyCancellationError(Exception):
+    pass
+
+
+class Task_Proxy(abc.ABC):
```
Comment (pfaion) [inline pupil_src/shared_modules/background_helper/base.py]: 1. Remove underscore from class names

2. Why is this not called `BaseTaskProxy`? Same with the 2 different implementations (threaded vs multiprocessing). I see that you are exposing them with different names through the module's `__init__.py`, why not name them differently in the first place? This makes it a lot easier to search for them as well.

File: pupil_src/shared_modules/background_helper/base.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self, name: str, generator, args=(), kwargs={}, **_):
+        self._name = name
+
+    @abc.abstractmethod
+    def fetch(self) -> typing.Iterator[typing.Any]:
+        """Fetches progress and available results from background"""
+        pass
+
+    @abc.abstractmethod
+    def cancel(self, timeout=1):
```
Comment (pfaion) [inline pupil_src/shared_modules/background_helper/base.py]: Add a comment for behavior of timeout.

**Comment by papr** [comment]
Similar to #1637 Surface Tracker is in a manageable state right now. Closing this PR until the situation changes.

**Pull request closed without merging**